girder/utility/assetstore_utilities.py

```diff
@@ -60,6 +59,15 @@ def getAssetstoreAdapter(assetstore, instance=True):
         return cls
 
 
+def setAssetstoreAdapter(storeType, cls):
+    global assetstoreTable
+
+    if storeType not in assetstoreTable:
+        raise Exception('Illegal assetstore type code: "%s"' % (storeType))
```

Comment:
Does that mean we'd dump the logic for that particular hook as well?

On Thu, Jun 9, 2016 at 3:00 PM Zach Mullen <notifications@github.com> wrote:

> In girder/utility/assetstore_utilities.py
> <https://github.com/girder/girder/pull/1372#discussion_r66501460>:
>
> > @@ -60,6 +59,15 @@ def getAssetstoreAdapter(assetstore, instance=True):
> >          return cls
> >
> >
> > +def setAssetstoreAdapter(
>
> storeType, cls):
> > +    global assetstoreTable
> > +
> > +    if storeType not in assetstoreTable:
> > +        raise Exception('Illegal assetstore type code: "%s"' % (storeType))
>
> I think this is actually fine, I'd like for plugins to be able to call
> this as a way to register new assetstore adapters as opposed to using the
> old assetstore.adapter.get hook.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/girder/girder/pull/1372/files/91b74f1930d5a4c381b47090446bc4fccef8f68b#r66501460>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACxNJE-r1QPQ7wDG-YaVvn-qrQLlPe8_ks5qKGKxgaJpZM4IyQ3c>
> .
>